### PR TITLE
Admin: Fix admin JS

### PR DESCRIPTION
**Pull request by tulimaki**
Admin: Fix BasePopupChoiceWidget filter
Admin: drop strict-mode from dashboard
Admin js-packages: Enable es6 for picotable

Refs SHOOP-1839

### CODE REVIEW

File: shoop/admin/gulp_bits/js-packages.js [hunk 1]
```diff
@@ -67,6 +68,7 @@ module.exports = {
         ]
     },
     "order-creator": {
+        "es6": true,
```
Comment (akx) [inline shoop/admin/gulp_bits/js-packages.js]: The es6 switch has no effect in webpack bundles.

**Comment by akx** [comment]
Re the Dashboard change: `const` really shouldn't be used in non-transpiled code at all due to flaky cross-browser support at all. Would be better to enable `es6` for the file, and change the logic so Chartist (and likely all `bower_components`) don't get es6 transpiled. http://caniuse.com/#feat=const

**Comment by suutari-ai** [comment]
Commit 8abf97a:
> Admin: Fix BasePopupChoiceWidget filter
>
> Since changes to filter in popup widgets we can't pass
> data-filter="None" to browse widget anymore since it is not
> JSON serializable.

The contents of this commit is OK, but the commit message is somehow misleading. I see no JSON serialization here, but does that even matter. There is no point in rendering data-filter attribute if there is no filter set anyway (no matter if it is JSON serialized or not). (Maybet that's what @akx meant too?)

**Comment by suutari-ai** [comment]
Re @akx comment:
> Would be better to enable es6 for the file, and change the logic so Chartist (or anything in bower_components) don't get es6 transpiled.

I agree on this. @tulimaki/@akx: Would this be hard to do?

**Comment by akx** [comment]
@suutari-ai: very easy. Just a gulp-if filter in the JS metatask.

File: shoop/admin/forms/widgets.py [hunk 2]
```diff
@@ -75,14 +75,17 @@ def render(self, name, value, attrs=None):
...
+
         return mark_safe("<div %(attrs)s>%(content)s</div>" % {
-            "attrs": flatatt({
-                "class": "browse-widget %s-browse-widget" % self.browse_kind,
-                "data-browse-kind": self.browse_kind,
-                "data-clearable": self.clearable,
-                "data-empty-text": self.empty_text,
-                "data-filter": self.filter
-            }),
+            "attrs": flatatt(browser_attrs),
```
Comment (akx) [inline shoop/admin/forms/widgets.py]: It'd be better to just use `flatatt_filter` (in `shoop.admin.toolbar`), which automatically elides falsy values.

**Comment by akx** [comment]
@suutari-ai, @tulimaki: re https://github.com/shoopio/shoop/commit/8abf97a61b7c8a42e97f84499ff408450bc2221e --

I also wanted to know what JSON serialization is suddenly having trouble.

But as mentioned in the inline code comment just now, we have an utility for eliding the attr, `flatatt_filter` :)

**Comment by tulimaki** [comment]
@akx https://github.com/shoopio/shoop/commit/d53b733b6c058a45aba91fa525c7ef2453763063

**Comment by suutari-ai** [comment]
@tulimaki: Include the commit ref about the JSON serialization having trouble to the commit message...

**Comment by akx** [comment]
@tulimaki,right, so that would also be fixed by turning `"data-filter": self.filter` to `"data-filter": self.filter or ""`, to make it serialize into an empty attribute. But yeah, `flatatt_filter`.

**Comment by suutari-ai** [comment]
@tulimaki Superb! Thanks!

**Pull request merged**